--- origsrc/srtp/Makefile.in	2009-06-15 11:25:50.685912000 -0500
+++ src/srtp/Makefile.in	2009-06-15 11:46:36.608449500 -0500
@@ -26,7 +26,7 @@ runtest: build_table_apps test
 	test/roc_driver$(EXE) -v >/dev/null
 	test/replay_driver$(EXE) -v >/dev/null
 	test/dtls_srtp_driver$(EXE) >/dev/null
-	cd test; ./rtpw_test.sh >/dev/null	
+	cd test; $(srcdir)/test/rtpw_test.sh >/dev/null	
 	@echo "libsrtp test applications passed."
 	$(MAKE) -C crypto runtest
 
@@ -60,6 +60,7 @@ top_builddir = @top_builddir@
 VPATH = @srcdir@
 prefix = @prefix@
 exec_prefix = @exec_prefix@
+bindir = @bindir@
 includedir = @includedir@
 libdir = @libdir@
 
@@ -105,6 +106,9 @@ libsrtp.a: $(srtpobj) $(cryptobj) $(gdoi
 	ar cr libsrtp.a $^
 	$(RANLIB) libsrtp.a
 
+cygsrtp-1.dll: $(srtpobj) $(cryptobj) $(gdoi)
+	$(CC) $(LDFLAGS) -shared -Wl,--out-implib,libsrtp.dll.a -o $@  $^
+
 # libcryptomath.a contains general-purpose routines that are used to
 # generate tables and verify cryptoalgorithm implementations - this
 # library is not meant to be included in production code
@@ -194,9 +198,14 @@ install:
 	fi
 	$(INSTALL) -d $(DESTDIR)$(includedir)/srtp
 	$(INSTALL) -d $(DESTDIR)$(libdir)
-	cp include/*.h $(DESTDIR)$(includedir)/srtp  
-	cp crypto/include/*.h $(DESTDIR)$(includedir)/srtp
+	cp $(srcdir)/include/*.h $(DESTDIR)$(includedir)/srtp  
+	cp $(srcdir)/crypto/include/*.h crypto/include/config.h $(DESTDIR)$(includedir)/srtp
 	if [ -f libsrtp.a ]; then cp libsrtp.a $(DESTDIR)$(libdir)/; fi
+	if [ -f cygsrtp-1.dll ]; then \
+		$(INSTALL) -d $(DESTDIR)$(bindir); \
+		cp cygsrtp-1.dll $(DESTDIR)$(bindir)/; \
+		cp libsrtp.dll.a $(DESTDIR)$(libdir)/; \
+	fi
 
 uninstall:
 	rm -rf $(DESTDIR)$(includedir)/srtp
